lib/CloudinaryPlugin.js

```diff
@@ -0,0 +1,335 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2019 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const Cloudinary = require('cloudinary').v2;
+
+class CloudinaryPlugin {
+
+  constructor() {
+    this.context = null;
+  }
+
+  init(config, context) {
+    this.context = context;
+
+    this.controllers = {
+      assetsController: {
+        search: 'search',
+        rename: 'rename',
+        destroy: 'destroy',
+        transform: 'transform'
+      },
+
+      tagsController: {
+        add_tag: 'add_tag',
+        remove_tag: 'remove_tag',
+        remove_all_tags: 'remove_all_tags',
+        replace_tag: 'replace_tag'
+      },
+    };
+
+    this.routes = [
+      {
+        verb: 'get',
+        url: '/assets/:expression',
+        controller: 'assetsController',
+        action: 'search'
+      },
+      {
+        verb: 'get',
+        url: '/assets/transform',
+        controller: 'assetsController',
+        action: 'transform'
+      },
+      {
+        verb: 'put',
+        url: '/assets',
+        controller: 'assetsController',
+        action: 'rename'
+      },
+      {
+        verb: 'delete',
+        url: '/assets',
+        controller: 'assetsController',
+        action: 'destroy'
+      },
+      {
+        verb: 'post',
+        url: '/tags',
+        controller: 'tagsController',
+        action: 'add_tag'
+      },
+      {
+        verb: 'put',
+        url: '/tags',
+        controller: 'tagsController',
+        action: 'replace_tag'
+      },
+      {
+        verb: 'delete',
+        url: '/tags',
+        controller: 'tagsController',
+        action: 'remove_tag'
+      },
+      {
+        verb: 'delete',
+        url: '/tags/removeAll',
```

Comment:
Use snake_case in url
```suggestion
        url: '/tags/remove_all',
```